lib/modules/manager/pep621/processors/uv.ts

```diff
@@ -175,6 +179,36 @@ export class UvProcessor implements PyProjectProcessor {
   }
 }
 
+function applyGitSource(dep: PackageDependency, depSource: UvGitSource): void {
+  const { git, rev, tag, branch } = depSource;
+  if (tag) {
+    const { source, full_name: repo } = parseGitUrl(git);
+    if (source === 'github.com') {
```

Comment:
```suggestion
   const platform = detectPlatform(source);
    if (platform === 'github') {
    dep.registryUrls = [`${test.protocol}://${test.source}`];
```
That way self hosted instances will be supported.

Same then for Gitlab.

Reply:
Having a `registryUrls` value will set [`UV_EXTRA_INDEX_URL`](https://github.com/renovatebot/renovate/pull/31928/files/44b41bd4585ea2d809538b935d2a16a99a0ad045#diff-d8da9aafff81cdba1a3495ed391d9536f840192d4d417ee7fea96447ca421c2fR242-R265) and I'm not sure we want that (I'm no more able to resolve some dependencies https://github.com/NextFire/renovate-repro-uv-git-rev/pull/19#issuecomment-2412322530)

Reply:
UV_EXTRA_INDEX_URL should filter by datasource in that case, based on the 

```typescript
  const pyPiRegistryUrls = deps
    .filter((dep) => dep.datasource === PypiDatasource.id)
    .map((dep) => dep.registryUrls)
    .flat()
  const registryUrls = new Set(pyPiRegistryUrls);
```

Reply:
This works, thank you!